exercises/armstrong-numbers/src/example.c

```diff
@@ -0,0 +1,30 @@
+#include "armstrong_numbers.h"
+#include <math.h>
+
+static uint16_t getDigitCount(uint32_t candidate)
+{
+   uint16_t digit_count = 0;
+   while (candidate > 0) {
+      candidate /= 10;
+      ++digit_count;
+   }
+   return digit_count;
+}
+
+bool isArmstrongNumber(uint32_t candidate)
+{
+   bool result = false;
+   if (candidate < 10) {
+      result = true;
+   } else if (candidate >= 100) {
```

Comment:
Are these cases necessary? Seems like it complicates the solution without any real benefit.

Reply:
All 1 digit numbers are Armstrong numbers but no 2 digit ones are. The cases shortcut the while loop which are unnecessary for these cases.
Technically, yes the solution would work without them... but the logic is evident from the test cases and exercise description.